model-server-openapi/specifications/model-server.yaml

```diff
@@ -13,481 +13,700 @@ info:
-            type: string
-        - name: Connection
-          in: "header"
-          required: true
-          description: "Websocket Connection parameter"
-          schema:
-            type: string
-        - name: Upgrade
-          in: "header"
-          required: true
-          description: "Websocket Upgrade parameter"
-          schema:
-            type: string
-        - name: Sec-WebSocket-Key
-          in: "header"
-          required: true
-          description: "Websocket Sec-WebSocket-Key parameter"
-          schema:
-            type: string
-      responses:
-        "403":
-          $ref: '#/components/responses/403'
-        "401":
-          $ref: '#/components/responses/401'
-        "500":
-          $ref: '#/components/responses/500'
-        "101":
-          description: "Switching Protocols"
-          headers:
-            Connection:
-              required: true
-              schema:
-                type: string
-            Upgrade:
-              required: true
-              schema:
-                type: string
-            Sec-WebSocket-Accept:
-              required: true
+          $ref: "#/components/responses/VersionDelta"
+        "404":
+          description: Repository, branch or `from` hash does not exist.
+          content:
+            application/problem+json:
               schema:
-                type: string
-  /v2/repositories/{repository}/branches/{branch}/poll:
-    get:
-      operationId: pollRepositoryBranch
-      parameters:
-        - name: lastKnown
-          in: "query"
-          required: false
-          schema:
-            type: string
-        - name: repository
-          in: "path"
-          required: true
-          schema:
-            type: string
-        - name: branch
-          in: "path"
-          required: true
-          schema:
-            type: string
+                $ref: "#/components/schemas/Problem"
+        default:
+          $ref: "#/components/responses/Problem"
+
+  # TODO specify input and output formats
+  /repositories/{repository}/branches/{branch}/query:
+    parameters:
+      - $ref: "#/components/parameters/RepositoryPath"
+      - $ref: "#/components/parameters/BranchPath"
+
+    post:
+      operationId: postRepositoryBranchQuery
+      tags:
+        - ModelQl access
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
       responses:
-        "403":
-          $ref: '#/components/responses/403'
-        "401":
-          $ref: '#/components/responses/401'
-        "500":
-          $ref: '#/components/responses/500'
         "200":
-          $ref: '#/components/responses/versionDelta'
-  /v2/repositories/{repository}/branches/{branch}/pollHash:
+          $ref: '#/components/responses/200json'
+        default:
+          $ref: '#/components/responses/Problem'
+
+  /repositories/{repository}/versions/{versionHash}/delta:
```

Comment:
I'm not sure yet, how I feel about including the `repository` path parameter in version related routes.
It makes sense structurally, but our versionHashes are unique across repositories, so given a `versionHash` one would first have to determine which `repository` it belongs to, to be able to query it.
This somewhat feels like a step backwards.